Windows CI failure: restore wallet test case related to pruned blockchain in `wallet_assumeutxo.py` is failing.

ISTM that the [Windows, ucrt, test cross-built CI failure](https://github.com/bitcoin/bitcoin/actions/runs/21172425203/job/60894641918?pr=33014) is unrelated, a restore wallet test case related to pruned blockchain in `wallet_assumeutxo.py` is failing.

<details open>
<summary>From the logs:</summary>

```zsh
2026-01-20T13:41:07.2385503Z 2026-01-20T13:41:04.152353Z TestFramework (INFO): Ensuring wallet can't be restored from a backup that was created before the pruneheight (pruned node)
2026-01-20T13:41:07.2386535Z 
2026-01-20T13:41:07.2386839Z 2026-01-20T13:41:06.479889Z TestFramework (ERROR): Unexpected exception
2026-01-20T13:41:07.2387306Z 
2026-01-20T13:41:07.2387503Z Traceback (most recent call last):
2026-01-20T13:41:07.2387815Z 
2026-01-20T13:41:07.2388269Z   File "D:\a\bitcoin\bitcoin\test\functional\test_framework\util.py", line 166, in try_rpc
2026-01-20T13:41:07.2388901Z 
2026-01-20T13:41:07.2389049Z     fun(*args, **kwds)
2026-01-20T13:41:07.2389318Z 
2026-01-20T13:41:07.2389432Z     ~~~^^^^^^^^^^^^^^^
2026-01-20T13:41:07.2389660Z 
2026-01-20T13:41:07.2390180Z   File "D:\a\bitcoin\bitcoin\test\functional\test_framework\coverage.py", line 50, in __call__
2026-01-20T13:41:07.2390808Z 
2026-01-20T13:41:07.2391156Z     return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
2026-01-20T13:41:07.2391651Z 
2026-01-20T13:41:07.2392093Z   File "D:\a\bitcoin\bitcoin\test\functional\test_framework\authproxy.py", line 156, in __call__
2026-01-20T13:41:07.2392771Z 
2026-01-20T13:41:07.2392996Z     raise JSONRPCException(response['error'], status)
2026-01-20T13:41:07.2393424Z 
2026-01-20T13:41:07.2395532Z test_framework.authproxy.JSONRPCException: filesystem error: cannot remove: The process cannot access the file because it is being used by another process [D:\a\_temp\test_runner_₿_🏃_20260120_132914\wallet_assumeutxo_56\node3\regtest\w2\wallet.dat] (-1)
2026-01-20T13:41:07.2397271Z 
2026-01-20T13:41:07.2397277Z 
2026-01-20T13:41:07.2397281Z 
2026-01-20T13:41:07.2397613Z During handling of the above exception, another exception occurred:
2026-01-20T13:41:07.2398102Z 
2026-01-20T13:41:07.2398107Z 
2026-01-20T13:41:07.2398112Z 
2026-01-20T13:41:07.2398334Z Traceback (most recent call last):
2026-01-20T13:41:07.2398622Z 
2026-01-20T13:41:07.2399135Z   File "D:\a\bitcoin\bitcoin\test\functional\test_framework\test_framework.py", line 142, in main
2026-01-20T13:41:07.2399796Z 
2026-01-20T13:41:07.2399922Z     self.run_test()
2026-01-20T13:41:07.2400161Z 
2026-01-20T13:41:07.2400286Z     ~~~~~~~~~~~~~^^
2026-01-20T13:41:07.2400474Z 
2026-01-20T13:41:07.2400948Z   File "D:\a\bitcoin\bitcoin/test/functional/wallet_assumeutxo.py", line 267, in run_test
2026-01-20T13:41:07.2401587Z 
2026-01-20T13:41:07.2401827Z     self.test_restore_wallet_pruneheight(n3)
2026-01-20T13:41:07.2402338Z 
2026-01-20T13:41:07.2402476Z     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^
2026-01-20T13:41:07.2402806Z 
2026-01-20T13:41:07.2403384Z   File "D:\a\bitcoin\bitcoin/test/functional/wallet_assumeutxo.py", line 99, in test_restore_wallet_pruneheight
2026-01-20T13:41:07.2404151Z 
2026-01-20T13:41:07.2404691Z     assert_raises_rpc_error(-4, error_message, n3.restorewallet, "w2", "backup_w2.dat")
2026-01-20T13:41:07.2405294Z 
2026-01-20T13:41:07.2405553Z     ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-01-20T13:41:07.2405970Z 
2026-01-20T13:41:07.2406496Z   File "D:\a\bitcoin\bitcoin\test\functional\test_framework\util.py", line 157, in assert_raises_rpc_error
2026-01-20T13:41:07.2407238Z 
2026-01-20T13:41:07.2407544Z     assert try_rpc(code, message, fun, *args, **kwds), "No exception raised"
2026-01-20T13:41:07.2408097Z 
2026-01-20T13:41:07.2408261Z            ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-01-20T13:41:07.2408598Z 
2026-01-20T13:41:07.2409048Z   File "D:\a\bitcoin\bitcoin\test\functional\test_framework\util.py", line 170, in try_rpc
2026-01-20T13:41:07.2409645Z 
2026-01-20T13:41:07.2410041Z     raise AssertionError("Unexpected JSONRPC error code %i" % e.error["code"])
2026-01-20T13:41:07.2410583Z 
2026-01-20T13:41:07.2410797Z AssertionError: Unexpected JSONRPC error code -1
2026-01-20T13:41:07.2411215Z 
2026-01-20T13:41:07.2411925Z 2026-01-20T13:41:06.484138Z TestFramework (INFO): Not stopping nodes as test failed. The dangling processes will be cleaned up later.
2026-01-20T13:41:07.2412795Z 
2026-01-20T13:41:07.2413676Z 2026-01-20T13:41:06.484270Z TestFramework (WARNING): Not cleaning up dir D:\a\_temp\test_runner_₿_🏃_20260120_132914\wallet_assumeutxo_56
2026-01-20T13:41:07.2414525Z 
2026-01-20T13:41:07.2415646Z 2026-01-20T13:41:06.484375Z TestFramework (ERROR): Test failed. Test logging available at D:\a\_temp\test_runner_₿_🏃_20260120_132914\wallet_assumeutxo_56/test_framework.log
2026-01-20T13:41:07.2416737Z 

2026-01-20T13:42:33.3464929Z wallet_assumeutxo.py                                                             | ✖ Failed  | 8 s
2026-01-20T13:42:33.3465209Z 
2026-01-20T13:42:33.3465460Z ALL                                                                              | ✖ Failed  | 2893 s (accumulated) 
2026-01-20T13:42:33.3465784Z Runtime: 796 s

```
</details>

_Originally posted by @rkrux in https://github.com/bitcoin/bitcoin/issues/33014#issuecomment-3773073528_

--- a/test/functional/wallet_assumeutxo.py
+++ b/test/functional/wallet_assumeutxo.py
@@ -96,7 +96,7 @@ class AssumeutxoTest(BitcoinTestFramework):
         assert_equal(n3.pruneblockchain(FINAL_HEIGHT), 298)  # 298 is the height of the last block pruned (pruneheight 299)
         error_message = "Wallet loading failed. Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of a pruned node)"
         # This backup (backup_w2.dat) was created at height 199, so it can't be restored in a node with a pruneheight of 299
-        assert_raises_rpc_error(-4, error_message, n3.restorewallet, "w2", "backup_w2.dat")
+        assert_raises_rpc_error(-4, error_message, n3.restorewallet, "w2_pruneheight", "backup_w2.dat")
 
         self.log.info("Ensuring wallet can be restored from a backup that was created at the pruneheight (pruned node)")
         # This backup (backup_w.dat) was created at height 299, so it can be restored in a node with a pruneheight of 299

